addons/im_livechat/static/src/embed/common/livechat_button.js

```diff
@@ -52,16 +52,29 @@ export class LivechatButton extends Component {
             animateNotification: !(
                 this.livechatService.thread || this.livechatService.shouldRestoreSession
             ),
+            hasAlreadyMovedOnce: false,
         });
         useMovable({
             cursor: "grabbing",
             ref: this.ref,
             elements: ".o-livechat-LivechatButton",
             onDrop: ({ top, left }) => {
+                this.state.hasAlreadyMovedOnce = true;
                 this.position.left = `${left}px`;
                 this.position.top = `${top}px`;
             },
         });
+        useExternalListener(document.body, "scroll", this._onScroll, { capture: true });
+    }
+
+    _onScroll(ev) {
+        if (!this.ref.el || this.state.hasAlreadyMovedOnce) {
+            return;
+        }
+        const container = ev.target;
+        container.scrollHeight - container.scrollTop === container.clientHeight
+            ? (this.position.top = `calc(93% - ${LIVECHAT_BUTTON_SIZE}px)`)
+            : (this.position.top = `calc(97% - ${LIVECHAT_BUTTON_SIZE}px)`);
```

Comment:
I find it more natural this way. We first read the assignment and understand what these three lines are about. Then we can focus on what's being assigned. Moreover, we avoid the assignment repetition.
```suggestion
        this.position.scrollTop = container.scrollHeight - container.scrollTop === container.clientHeight
            ? `calc(93% - ${LIVECHAT_BUTTON_SIZE}px)`
            : `calc(97% - ${LIVECHAT_BUTTON_SIZE}px)`;
```

Reply:
Oh absolutly mb here, commiting your suggestion

Reply:
You reverted the changes to the commit message while doing so (period + line break between paragraphs).

Reply:
definitely not aware this morning sorry :(

Reply:
Sorry about that, it's actually `position.top` not `position.scrollTop`.

Reply:
and of course, my cache tricked me and I thought it worked when testing it

doing the change